From: Stefan Wahren <wahrenst@gmx.net>
Date: Wed, 25 Dec 2019 15:43:41 +0100
Subject: ARM: dts: bcm2838-rpi-4-b: Backport BT part from upstream

The CYW43455 on the Raspberry Pi 4 doesn't use an external pin as lower
power clock anymore. So drop the GPIO clock from pinctrl. While we are at
this add the missing declaration of hardware flow control.

Signed-off-by: Stefan Wahren <wahrenst@gmx.net>
---
 arch/arm/boot/dts/bcm2838-rpi-4-b.dts | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/bcm2838-rpi-4-b.dts b/arch/arm/boot/dts/bcm2838-rpi-4-b.dts
index 35502763882e..e65f924e1fd2 100644
--- a/arch/arm/boot/dts/bcm2838-rpi-4-b.dts
+++ b/arch/arm/boot/dts/bcm2838-rpi-4-b.dts
@@ -101,7 +101,8 @@
 /* uart0 communicates with the BT module */
 &uart0 {
 	pinctrl-names = "default";
-	pinctrl-0 = <&uart0_ctsrts_gpio30 &uart0_gpio32 &gpclk2_gpio43>;
+	pinctrl-0 = <&uart0_ctsrts_gpio30 &uart0_gpio32>;
+	uart-has-rtscts;
 	status = "okay";
 
 	bluetooth {
